### PR TITLE
Rnd 754/setup project

### PR DESCRIPTION
**Pull request by SzajekSzymon**
## Description

Setup and configuration project to development

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the [code style](/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---
I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

**Comment by tomaszmichalak** [comment]
@SzajekSzymon Please fill PR details: description etc.

**Comment by tomaszmichalak** [comment]
@SzajekSzymon Please rebase and squash all commits. Please add commit message in the form of: "#ISSUE description".

File: src/js/components/Graphs/exampleGraph.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+// THIS IS AN EXAMPLE COMPONENT . SHOULD REMOVE IN FUTURE.
```
Comment (tomaszmichalak) [inline src/js/components/Graphs/exampleGraph.jsx]: Please remove comments.

File: .babelrc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (tomaszmichalak) [inline .babelrc]: Can we provide some license headers where it is possible?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "knotx-extenstion",
```
Comment (Shavrin) [inline package.json]:
```suggestion
  "name": "knotx-extension",
```

File: src/js/helpers/nodesHelper.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  let isBetweenComments = false;
+  let curComment = '';
+  // eslint-disable-next-line no-cond-assign
+  while ((curNode = iterator.nextNode())) {
+    if (
+      curNode.nodeType === COMMENT_NODE_CODE
+          && curNode.data.indexOf('data-knotx-id') !== -1
+    ) {
+      isBetweenComments = !isBetweenComments;
+      curComment = isBetweenComments ? curNode.data.match(/"([^']+)"/)[1] : '';
```
Comment (Skejven) [inline src/js/helpers/nodesHelper.js]: `curComment` - isn't that Fragment's ID? If yes, maybe variable name `currentFragmentId` would be more meaningful?

File: src/js/helpers/nodesHelper.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+    if (
+      isBetweenComments
+          && curNode.nodeType !== COMMENT_NODE_CODE
+          && curNode.nodeType === HTML_NODE_CODE
+    ) {
+      if (!COMMENTS.length) {
+        curNode.dataset.knotxId = curComment;
+        COMMENTS.push(curNode);
+      } else if (!COMMENTS[COMMENTS.length - 1].contains(curNode)) {
```
Comment (Skejven) [inline src/js/helpers/nodesHelper.js]: Looks like this condition and one from line `46` can be merged into one joined by `OR`. In both cases, the logic in the `if` body is exactly the same.
BTW, what does this condition check?

File: package.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  "name": "knotx-chrome-devtool-extension",
+  "version": "1.0.0",
+  "description": "",
+  "private": true,
+  "scripts": {
+    "dev": "webpack",
+    "watch": "watch 'npm run dev'"
+  },
+  "keywords": [],
+  "author": "",
```
[the remark this answers is not in the document]
Comment (tomaszmichalak) [reply package.json]: We can also define contributors soon. Those fields have syntax:
```
{
  "name" : "Barney Rubble",
  "email" : "b@rubble.com",
  "url" : "http://barnyrubble.tumblr.com/"
}
```

File: src/js/apps/App.jsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import HelloReact from '../components/helloReact';
```
Comment (tomaszmichalak) [inline src/js/apps/App.jsx]: Can we please remove sample content?

File: src/js/apps/App.jsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import ExampleGraph from '../components/Graphs/exampleGraph';
+
+const App = () => (
+  <div className="app">
+    <HelloReact />
```
Comment (tomaszmichalak) [inline src/js/apps/App.jsx]: Please remove this component.

File: src/js/helpers/nodesHelper.js [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export const knotxNodes = () => {
+  const rootElem = document.body;
+  const COMMENTS = [];
+  const COMMENT_NODE_CODE = 8;
+  const HTML_NODE_CODE = 1;
+  // Fourth argument, which is actually obsolete according to the DOM4 standard, is required in IE 11
```
[the remark this answers is not in the document]
Comment (SzajekSzymon) [reply src/js/helpers/nodesHelper.js]: Rewrite nodeHelper.js task, should be in another ticket I think. This code work, but its not beautiful. It's important to merge this PR to make continue developing possible

File: .eslintrc.json [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+            "jsx": true
+        },
+        "ecmaVersion": 2018,
+        "sourceType": "module"
+    },
+    "plugins": [
+        "react"
+    ],
+    "rules": {
+        "max-len": ["error", { "code": 120 }],
```
[the remark this answers is not in the document]
Comment (SzajekSzymon) [reply .eslintrc.json]: We use airbnb configuration for another rules. (probably most popular eslint configuration). You can find this in section "extends" line 9.

File: src/js/components/helloReact.jsx [hunk 11]
```diff
@@ -0,0 +1,25 @@
+/*
```
Comment (tomaszmichalak) [inline src/js/components/helloReact.jsx]: Do we need this component?

File: README.md [hunk 12]
```diff
@@ -25,12 +25,16 @@ Knot.x Fragments Chrome Extension reads data from the HTML response and loads
 fragment's debug data and visualize task details.
 
 ### How to use?
+1. go to the project folder
+2. run command: npm install
```
Comment (tomaszmichalak) [inline README.md]:
```suggestion
2. run command: `npm install`
```

File: README.md [hunk 13]
```diff
@@ -25,12 +25,16 @@ Knot.x Fragments Chrome Extension reads data from the HTML response and loads
...
+3. run command: npm run dev
```
Comment (tomaszmichalak) [inline README.md]:
```suggestion
3. run command: `npm run dev`
```

File: package-lock.json [hunk 14]
```diff
@@ -0,0 +1,8024 @@
+{
```
Comment (tomaszmichalak) [inline package-lock.json]: I think this file should not be stored in repository.

File: src/js/index.jsx [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+ * limitations under the License.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './apps/App';
+import '../sass/style.scss';
+import { knotxNodes } from './helpers/nodesHelper';
+
+chrome.devtools.panels.create('knotx', null, 'index.html');
```
Comment (tomaszmichalak) [inline src/js/index.jsx]: Do we need `index.html` file in codebase?

File: src/js/apps/App.jsx (hunk 7 above)
Comment (SzajekSzymon) [reply src/js/apps/App.jsx]: Ok. What about example graph?

File: package-lock.json (hunk 14 above)
Comment (SzajekSzymon) [reply package-lock.json]: this file should be stored in repository. Every developers should have one version of this file to check compatibility of packages.

**Pull request merged**